src/components/esn-app-grid/esn-app-grid.tsx

```diff
@@ -0,0 +1,80 @@
+import { Component, Prop, ComponentInterface, State, Host, h } from '@stencil/core';
+import fetchSvgIcon from '../../utils/fetch-icon';
+import { InternalItem } from './esn-app-grid.types';
+import { AppGridTogglerIcon } from './app-grid-icon/AppGridIcon';
+
+@Component({
+  tag: 'esn-app-grid',
+  styleUrl: 'esn-app-grid.scss',
+  shadow: true
+})
+export class EsnAppGrid implements ComponentInterface {
+  private popover: HTMLElement;
+
+  @State() internalItems: InternalItem[] = [];
+
+  /**
+   * A serialized JSON object that contains information about the available apps.
+   */
+  @Prop() serializedItems: string;
+
+  componentWillLoad() {
+    try {
+      if (!this.serializedItems) {
+        throw new Error("The 'serializedItems' attribute/property must be set to a serialized JSON");
+      }
+
+      const items = JSON.parse(this.serializedItems);
+
+      if (Array.isArray(items) && items.length) {
+        this.internalItems = items.map(({ name, url }) => ({
+          name,
+          url
+        }));
+      } else {
+        throw new Error("The parsed 'serializedItems' has an invalid JSON structure");
+      }
+
+      Promise.all(items.map(({ iconUrl }) => fetchSvgIcon(iconUrl))).then(iconContents => {
+        this.internalItems = iconContents.map((iconContent, index) => ({
+          name: this.internalItems[index].name,
+          url: this.internalItems[index].url,
+          iconContent
```

Comment:
this should be 
```js
iconContent: this.internalItems[index].icon.data
```

without the need to fetch the icons by url and looping.

by that i mean the icon fetch is not necessary since it's in the `SPA.icon.data`

Reply:
This won't be necessary as we will include all the icons for a fixed set of applications in the web component itself.